core/src/main/scala/chisel3/BitsImpl.scala

```diff
@@ -558,7 +556,7 @@ private[chisel3] trait AsyncResetImpl extends Element { self: AsyncReset =>
 
   override def toString: String = stringAccessor("AsyncReset")
 
-  def cloneType: this.type = AsyncReset().asInstanceOf[this.type]
+  override def _cloneType: Data = AsyncReset()
```

Comment:
```suggestion
  override def _cloneType: AsyncReset = AsyncReset()
```

? perhaps? Why is this one different than the others